pkg/query-service/agentConf/manager.go

```diff
@@ -56,7 +56,7 @@ func GetConfigHistory(ctx context.Context, typ ElementTypeDef) ([]ConfigVersion,
 }
 
 // StartNewVersion launches a new config version for given set of elements
-func StartNewVersion(ctx context.Context, eleType ElementTypeDef, elementIds []string) (*ConfigVersion, error) {
+func StartNewVersion(ctx context.Context, userId string, eleType ElementTypeDef, elementIds []string) (*ConfigVersion, error) {
```

Comment:
@nityanandagohain  this is a new change to capture user Id. The context requires more thought, hence I am opting out of it for now

Reply:
cool.